monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/ReferenceDataController.java

```diff
@@ -19,59 +19,383 @@
+    }
+
+    @RolesAllowed(SecurityRoles.VIEWER)
+    @RequestMapping(value = "/dictionary/{name}", method = { RequestMethod.GET, RequestMethod.HEAD },
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public DictionaryModel getDictionary(@PathVariable("tenant") final String tenant,
+            @PathVariable("name") final String name, final HttpServletRequest request) {
+        final Dictionary dictionary = getReferenceDataCatalog(tenant).getDictionary(name);
+
+        if (dictionary == null) {
+            throw new NoSuchResourceException();
+        }
+
+        if (request.getMethod().equals(RequestMethod.HEAD.name())) {
+            return null;
+        }
+
+        final List<String> entries =
+                dictionary.openConnection(_contextFactory.getContext(tenant).getConfiguration()).stream()
+                        .collect(Collectors.toList());
+        return new DictionaryModel(name, entries, dictionary.isCaseSensitive());
+    }
+
+    @RolesAllowed(SecurityRoles.CONFIGURATION_EDITOR)
+    @RequestMapping(value = "/dictionary/{name}", method = { RequestMethod.PUT },
+            consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity putDictionary(@PathVariable("tenant") final String tenant,
+            @PathVariable("name") final String name, @RequestBody final DictionaryModel dictionaryModel) {
+        final TenantContext context = _contextFactory.getContext(tenant);
+        final FileResource resource = new FileResource(getResourceFile(context, name));
+        resource.write(
+                stream -> stream.write((Joiner.on("\n").join(dictionaryModel.getEntries()) + "\n").getBytes("UTF-8")));
+        final TextFileDictionary dictionary =
+                new TextFileDictionary(name, resource.getFile().getAbsolutePath(), "UTF-8",
+                        dictionaryModel.isCaseSensitive());
+
+        _referenceDataDao.removeDictionary(context, name);
+        _referenceDataDao.addDictionary(context, dictionary);
+        return ResponseEntity.created(ServletUriComponentsBuilder.fromCurrentRequestUri().build().toUri()).build();
+    }
+
+    @RolesAllowed(SecurityRoles.CONFIGURATION_EDITOR)
+    @RequestMapping(value = "/dictionary/{name}", method = { RequestMethod.PUT },
             consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
-    public String uploadReferenceData(@PathVariable("tenant") final String tenant,
-            @RequestParam("file") final MultipartFile file) {
-        String status = "success";
-
-        try {
-            processUploadedFile(tenant, file.getInputStream());
-        } catch (final Exception e) {
-            status = "failure";
+    public ResponseEntity putDictionaryFile(@PathVariable("tenant") final String tenant,
+            @PathVariable("name") final String name, @RequestParam("file") final MultipartFile sourceMultipartFile,
+            @RequestParam(value="casesensitive", defaultValue="true") final boolean isCaseSensitive,
```

Comment:
WhitespaceAround: '=' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)